### ISSUE TITLE
[BUG] OpenAi no longer works with custom models

### ISSUE THREAD

**Issue by jmolbach** [langchain4j/langchain4j#223]
**Describe the bug**
OpenAi Tokenizer fails in version 0.23.0 for custom trained models

**Log and Stack trace**
java.lang.IllegalArgumentException: Model 'ft:gpt-3.5-turbo-0613:papirfly-as::7vk6cwK5' is unknown to jtokkit
	at dev.langchain4j.internal.Exceptions.illegalArgument(Exceptions.java:6)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
	*__checkpoint ⇢ Handler com.papirfly.openaiapi.controller.OpenAiTextGenerationController#textGenerateStream(int, String, String, TextGenerationInput) [DispatcherHandler]
	*__checkpoint ⇢ org.springframework.web.filter.reactive.ServerHttpObservationFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ com.papirfly.starter.filter.RewriteWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ HTTP POST "/v1/text/generate-stream/10930" [ExceptionHandlingWebHandler]
Original Stack Trace:
		at dev.langchain4j.internal.Exceptions.illegalArgument(Exceptions.java:6)
		at dev.langchain4j.model.openai.OpenAiTokenizer.lambda$unknownModelException$0(OpenAiTokenizer.java:148)
		at java.base/java.util.Optional.orElseThrow(Optional.java:403)
		at dev.langchain4j.model.openai.OpenAiTokenizer.estimateTokenCountInText(OpenAiTokenizer.java:37)
		at dev.langchain4j.model.openai.OpenAiTokenizer.estimateTokenCountInMessage(OpenAiTokenizer.java:45)
		at dev.langchain4j.model.openai.OpenAiTokenizer.estimateTokenCountInMessages(OpenAiTokenizer.java:81)
		at dev.langchain4j.model.openai.OpenAiStreamingChatModel.generate(OpenAiStreamingChatModel.java:113)
		at dev.langchain4j.model.openai.OpenAiStreamingChatModel.generate(OpenAiStreamingChatModel.java:89)
		at dev.langchain4j.service.AiServiceTokenStream$AiServiceOnStart.start(AiServiceTokenStream.java:78)
		at com.papirfly.openaiapi.facade.OpenAiTextGenerationFacade$textGenerateStream$1.invoke(OpenAiTextGenerationFacade.kt:43)
		at com.papirfly.openaiapi.facade.OpenAiTextGenerationFacade$textGenerateStream$1.invoke(OpenAiTextGenerationFacade.kt:25)
		at com.papirfly.openaiapi.facade.OpenAiTextGenerationFacade.textGenerateStream$lambda$0(OpenAiTextGenerationFacade.kt:25)
		at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:386)
		at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onNext(FluxSwitchIfEmpty.java:74)
		at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onNext(FluxPeekFuseable.java:210)
		at reactor.core.publisher.MonoCompletionStage$MonoCompletionStageSubscription.apply(MonoCompletionStage.java:120)
		at reactor.core.publisher.MonoCompletionStage$MonoCompletionStageSubscription.apply(MonoCompletionStage.java:64)
		at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
		at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
		at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
		at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
		at software.amazon.awssdk.core.internal.http.pipeline.stages.AsyncApiCallMetricCollectionStage.lambda$execute$0(AsyncApiCallMetricCollectionStage.java:56)
		at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
		at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
		at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
		at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
		at software.amazon.awssdk.core.internal.http.pipeline.stages.AsyncApiCallTimeoutTrackingStage.lambda$execute$2(AsyncApiCallTimeoutTrackingStage.java:69)
		at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
		at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
		at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
		at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
		at software.amazon.awssdk.core.internal.http.pipeline.stages.AsyncRetryableStage$RetryingExecutor.lambda$attemptExecute$1(AsyncRetryableStage.java:177)
		at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
		at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
		at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
		at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
		at software.amazon.awssdk.core.internal.http.pipeline.stages.MakeAsyncHttpRequestStage.lambda$null$0(MakeAsyncHttpRequestStage.java:105)
		at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
		at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
		at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
		at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
		at software.amazon.awssdk.core.internal.http.pipeline.stages.MakeAsyncHttpRequestStage.lambda$executeHttpRequest$3(MakeAsyncHttpRequestStage.java:163)
		at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
		at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
		at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
		at java.base/java.lang.Thread.run(Thread.java:833)

**To Reproduce**
Create a OpenAiStreamingChatModel like this:
val chatModel = OpenAiStreamingChatModel("${openAiApiProperties.apiUrl}/v1", openAiApiProperties.apiKey, openAiModelDynamo.openAiModelName,
                    openAiModelDynamo.temperature, openAiModelDynamo.topP, emptyList(), openAiModelDynamo.maxTokens, openAiModelDynamo.presencePenalty, openAiModelDynamo.frequencyPenalty,
                    Duration.ofSeconds(openAiModelDynamo.timeoutSeconds), Proxy.NO_PROXY, true, true)
...where the openAiModelName is a name for a custom trained model, like mine: ft:gpt-3.5-turbo-0613:papirfly-as::7vk6cwK5

**Expected behavior**
Expecting it to work also with custom models. And if Tokenizer requires base models to work, being able to specify that as a separate parameter?

**Please complete the following information:**
- LangChain4j version: e.g. 0.23.0
- Java version: Kotlin 1.8.22
- Spring Boot version (if applicable): e.g. 2.7.14

**Additional context**
Error was not present in 0.22, was introduced in 0.23

**Comment by langchain4j** [comment]
@jmolbach thanks for reporting!

**Comment by langchain4j** [comment]
@jmolbach [fixed](https://github.com/langchain4j/langchain4j/commit/d90715fab554df450d08fadf144bfd3b5e5ca587), will be in the next release

**Comment by jmolbach** [comment]
@langchain4j , thanks :-) Appreciated.

**Comment by jmolbach** [comment]
@langchain4j , btw any idea when that will be? Is there a PR or branch that needs merging?

**Comment by langchain4j** [comment]
@jmolbach sorry for the late reply! 0.24.0 was released on Nov 12, 0.25.0 will be released later today

**Comment by nhannht** [comment]
The bug is still here. Can you recheck it?

```text
java.lang.IllegalArgumentException: Model 'undi95/toppy-m-7b:free' is unknown to jtokkit
	at dev.langchain4j.internal.Exceptions.illegalArgument(Exceptions.java:19) ~[langchain4j-core-0.28.0.jar:na]
	at dev.langchain4j.model.openai.OpenAiTokenizer.lambda$unknownModelException$0(OpenAiTokenizer.java:266) ~[langchain4j-open-ai-0.28.0.jar:na]
	at java.base/java.util.Optional.orElseThrow(Optional.java:403) ~[na:na]
	at dev.langchain4j.model.openai.OpenAiTokenizer.estimateTokenCountInText(OpenAiTokenizer.java:56) ~[langchain4j-open-ai-0.28.0.jar:na]
	at dev.langchain4j.model.openai.OpenAiTokenizer.estimateTokenCountIn(OpenAiTokenizer.java:81) ~[langchain4j-open-ai-0.28.0.jar:na]
	at dev.langchain4j.model.openai.OpenAiTokenizer.estimateTokenCountInMessage(OpenAiTokenizer.java:66) ~[langchain4j-open-ai-0.28.0.jar:na]
	at dev.langchain4j.model.openai.OpenAiTokenizer.estimateTokenCountInMessages(OpenAiTokenizer.java:168) ~[langchain4j-open-ai-0.28.0.jar:na]
	at dev.langchain4j.model.openai.OpenAiStreamingChatModel.generate(OpenAiStreamingChatModel.java:137) ~[langchain4j-open-ai-0.28.0.jar:na]
	at dev.langchain4j.model.openai.OpenAiStreamingChatModel.generate(OpenAiStreamingChatModel.java:104) ~[langchain4j-open-ai-0.28.0.jar:na]

```


![image](https://github.com/langchain4j/langchain4j/assets/69144096/ff034b0d-c66c-4db7-8a05-b2e18c2c62a9)
